android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncPatientItemMapper.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.anc.ui.anccare.details
+
+import android.content.Context
+import com.google.android.fhir.logicalId
+import java.time.Instant
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+import org.hl7.fhir.r4.model.Patient
+import org.hl7.fhir.r4.model.codesystems.AdministrativeGender
+import org.smartregister.fhircore.anc.AncApplication
+import org.smartregister.fhircore.anc.R
+import org.smartregister.fhircore.anc.data.model.AncPatientItem
+import org.smartregister.fhircore.engine.data.domain.util.DomainMapper
+
+object AncPatientItemMapper : DomainMapper<Patient, AncPatientItem> {
+  private const val RISK = "risk"
+
+  override fun mapToDomainModel(dto: Patient): AncPatientItem {
+    val name = dto.extractName()
+    val gender = dto.extractGender(AncApplication.getContext()).first()
+    val age = dto.extractAge()
+    return AncPatientItem(
+      patientIdentifier = dto.logicalId,
+      name = name,
+      gender = gender.toString(),
+      age = age,
+      demographics = "$name, $gender, $age",
+      atRisk = dto.atRisk()
+    )
+  }
+
+  fun Patient.extractName(): String {
+    if (!hasName()) return ""
+    val humanName = this.name.firstOrNull()
+    return if (humanName != null) {
+      "${
+      humanName.given.joinToString(" ")
+      { it.toString().trim().toTitleCase() }
+      } ${humanName.family?.toTitleCase() ?: ""}"
+    } else ""
+  }
+
+  private fun String.toTitleCase() = replaceFirstChar {
+    if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
+  }
+
+  fun Patient.extractGender(context: Context) =
+    when (AdministrativeGender.valueOf(this.gender.name)) {
+      AdministrativeGender.MALE -> context.getString(R.string.male)
+      AdministrativeGender.FEMALE -> context.getString(R.string.female)
+      AdministrativeGender.OTHER -> context.getString(R.string.other)
+      AdministrativeGender.UNKNOWN -> context.getString(R.string.unknown)
+      AdministrativeGender.NULL -> ""
+    }
+
+  fun Patient.extractAge(): String {
+    if (!hasBirthDate()) return ""
```

Comment:
Can we refactor these implementations as extensions and keep them in the file named `PatientExtension`. Do the refactor too for `PatientItemMapper` in eir application module